neuro_extras/main.py

```diff
@@ -201,6 +231,158 @@ def image() -> None:
+        raise ValueError(f"Unsupported source URL scheme: {source_url.scheme}")
+    destination_url_type = UrlType.get_type(destination_url)
+    if destination_url_type == UrlType.UNSUPPORTED:
+        raise ValueError(
+            f"Unsupported destination URL scheme: {destination_url.scheme}"
+        )
+
+    if source_url_type == UrlType.CLOUD and destination_url_type == UrlType.CLOUD:
+        raise ValueError(
+            "This command can't be used to copy data between cloud providers"
+        )
+    if source_url_type == UrlType.STORAGE and destination_url_type == UrlType.STORAGE:
+        raise ValueError(
+            "This command can't be used to copy data between two storage locations"
+        )
+
+    if UrlType.STORAGE in (source_url_type, destination_url_type):
+        # raise ValueError("Storage operations are not supported yet")
+        async with neuro_api.get() as client:
+            data_copier = DataCopier(client)
+            if source_url_type == UrlType.STORAGE:
+                job = await data_copier.launch(
+                    storage_uri=source_url,
+                    src_uri=URL("/var/storage"),
+                    dst_uri=destination_url,
+                    unpack=unpack,
+                )
+            else:
+                job = await data_copier.launch(
+                    storage_uri=destination_url,
+                    src_uri=source_url,
+                    dst_uri=URL("/var/storage"),
+                    unpack=unpack,
+                )
+            while job.status == neuro_api.JobStatus.PENDING:
+                job = await client.jobs.status(job.id)
+                await asyncio.sleep(1.0)
+            async for chunk in client.jobs.monitor(job.id):
+                if not chunk:
+                    break
+                click.echo(chunk.decode(errors="ignore"), nl=False)
+            job = await client.jobs.status(job.id)
+            if job.status == neuro_api.JobStatus.FAILED:
+                logger.error("The copy job has failed due to:")
+                logger.error(f"  Reason: {job.history.reason}")
+                logger.error(f"  Description: {job.history.description}")
+                exit_code = job.history.exit_code
+                if exit_code is None:
+                    exit_code = EX_PLATFORMERROR
+                sys.exit(exit_code)
+            else:
+                logger.info("Successfully copied data")
+    else:
+        tmp_dir_name = tempfile.mkdtemp(dir=str(TEMP_UNPACK_DIR))
+        tmp_dst_url = URL.build(path=(tmp_dir_name + "/"))
+        print(f"temp destination url = {tmp_dst_url}")
+
+        await _nonstorage_cp(source_url, tmp_dst_url)
+        source_url = tmp_dst_url
+        # for clarity
+        source_url_type = UrlType.LOCAL
+
+        # at this point source is always local
+        if unpack:
+            file = Path(source_url.path)
+            if file.is_dir():
+                file = list(file.glob("*"))[0]
```

Comment:
If the source is a directory, we are trying to unarchive the first file in it?

Reply:
Right. That's the case when we download a file to a temp folder — our new source is the folder itself.